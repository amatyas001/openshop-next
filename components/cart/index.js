export * from './CartContent/CartContent';
export * from './CartAmount/CartAmount';
export * from './CartButton/CartButton';
export * from './CartItem/CartItem';
export * from './CartPanel/CartPanel';
