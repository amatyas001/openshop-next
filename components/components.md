Elements are composed by [Chakra-UI](https://chakra-ui.com) components and preserve the library features so [Style Props](https://chakra-ui.com/style-props) can be passed to them and accessibility remains optimized.

> _Style Props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components._

For more information visit the [Chakra UI](https://chakra-ui.com) documentation.

Components are divided into two main category: _wrappers_ and _single responsibility_ elements. Wrapper components does not contain business logic, however there are special cases like _handling lifted states to keep sensitive data locally_ and not to populate them to the Redux store which is persisted on the machine.

**State Dependencies** and **Wrapped Components** are documented in each description to give brief information about the context of posible interactions of the actual component. _Examples_ prepared with proper state and may render extra controls to test different cases with ease.
