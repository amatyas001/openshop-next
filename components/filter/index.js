export * from './FilterButton/FilterButton';
export * from './FilterColor/FilterColor';
export * from './FilterPrice/FilterPrice';
export * from './FilterName/FilterName';
export * from './FilterPanel/FilterPanel';
export * from './FilterReset/FilterReset';
