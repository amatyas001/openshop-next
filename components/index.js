export * from './shared';
export * from './cart';
export * from './payment';
