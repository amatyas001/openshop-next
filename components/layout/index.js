export * from './Content/Content';
export * from './Footer/Footer';
export * from './ProductsTopBar/ProductsTopBar';
export * from './Navbar/Navbar';
export * from './ScrollTop/ScrollTop';
