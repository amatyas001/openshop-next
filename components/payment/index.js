export * from './PaymentCancelled/PaymentCancelled';
export * from './PaymentConfirm/PaymentConfirm';
export * from './PaymentConfirmCard/PaymentConfirmCard';
export * from './PaymentConfirmControls/PaymentConfirmControls';
export * from './PaymentConfirmDetails/PaymentConfirmDetails';
export * from './PaymentError/PaymentError';
export * from './PaymentForm/PaymentForm';
export * from './PaymentFormControls/PaymentFormControls';
export * from './PaymentProgress/PaymentProgress';
export * from './PaymentReview/PaymentReview';
export * from './PaymentSuccess/PaymentSuccess';
