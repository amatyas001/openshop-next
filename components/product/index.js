export * from './ProductCard/ProductCard';
export * from './ProductControls/ProductControls';
export * from './ProductControlsColor/ProductControlsColor';
export * from './ProductControlsAmount/ProductControlsAmount';
export * from './ProductControlsButton/ProductControlsButton';
export * from './ProductControlsSize/ProductControlsSize';
