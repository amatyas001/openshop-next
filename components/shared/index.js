export * from './Button/Button';
export * from './Card/Card';
export * from './Field/Field';
export * from './Rating/Rating';
export * from './ScrollTop/ScrollTop';
export * from './Navbar/Navbar';
export * from './Footer/Footer';
export * from './Spinner/Spinner';
