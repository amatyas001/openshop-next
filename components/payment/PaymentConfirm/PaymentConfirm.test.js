import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { INITIAL_STATE } from '@app/config';
import { mockProductWithBuyAmount } from '@app/mocks';
import { paymentSuccess, paymentError } from '@app/lib/redux/actions';
import { PaymentConfirm } from '@app/components';

const mockStore = configureStore([]);

let tree;
let store;
let submit;
let card;
const state = {
  ...INITIAL_STATE,
  cart: mockProductWithBuyAmount(1),
  payment: {
    status: 'confirm',
    details: {
      name: 'mock_name',
      email: 'mock_email',
      phone: 'mock_phone',
      address: 'mock_address',
    },
    intent: {
      id: 'mock_id',
      secret: 'mock_secret',
    },
    token: 'token',
  },
};

describe('<PaymentConfirm />', () => {
  beforeAll(() => {
    store = mockStore(state);
    act(() => {
      tree = create(
        <Provider store={store}>
          <PaymentConfirm />
        </Provider>
      );
    });
    store.clearActions();
    submit = tree.root.findByProps({
      'data-testid': 'confirm-controls-confirm',
    });
    card = tree.root.findByProps({
      'data-testid': 'confirm-card-element',
    });
  });

  it('should render without props', () => {
    expect.assertions(1);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should not dispatch paymentSuccess', async () => {
    expect.assertions(1);
    act(() => {
      card.props.onChange({ complete: false });
    });
    await act(async () => {
      await submit.props.onClick();
    });
    expect(store.getActions()).toEqual([]);
  });

  it('should dispatch paymentSuccess', async () => {
    expect.assertions(1);
    act(() => {
      card.props.onChange({ complete: true });
    });
    await act(async () => {
      await submit.props.onClick();
    });
    expect(store.getActions()).toEqual([
      paymentSuccess({
        status: 'succeeded',
        ...store.getState().payment.intent,
      }),
    ]);
  });

  it('should dispatch paymentError', async () => {
    expect.assertions(1);
    // Mocking error response from Stripe API
    // See: __mocks__/@stripe/react-stripe-js/useStripe
    state.payment.intent.secret = 'error';
    store = mockStore(state);
    act(() => {
      tree = create(
        <Provider store={store}>
          <PaymentConfirm />
        </Provider>
      );
    });
    store.clearActions();
    submit = tree.root.findByProps({
      'data-testid': 'confirm-controls-confirm',
    });
    card = tree.root.findByProps({
      'data-testid': 'confirm-card-element',
    });
    act(() => {
      card.props.onChange({
        complete: true,
      });
    });
    await act(async () => {
      await submit.props.onClick();
    });
    expect(store.getActions()).toEqual([paymentError('mock_error')]);
  });

  it('should render with initial state', () => {
    expect.assertions(1);
    act(() => {
      tree = create(
        <Provider store={mockStore(INITIAL_STATE)}>
          <PaymentConfirm />
        </Provider>
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
