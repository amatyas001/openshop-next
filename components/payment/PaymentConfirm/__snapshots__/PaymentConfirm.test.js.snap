// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentConfirm /> should render without props 1`] = `
<div
  aria-hidden={false}
  className="css-1mi3tt8"
>
  <h2
    className="css-ijdyq2"
    data-testid="confirm-intent-name"
  >
    Hello, 
    mock_name
    !
  </h2>
  <p
    className="css-tmw1y6"
  >
    Please review your payment details and confirm order
  </p>
  <div
    className="css-r96hqv"
  >
    <p
      className="css-18poma1"
    >
      INTENT ID
    </p>
    <p
      className="css-1n4bukm"
      data-testid="confirm-intent-id"
    >
      mock_id
    </p>
    <p
      className="css-18poma1"
    >
      EMAIL
    </p>
    <p
      className="css-1n4bukm"
      data-testid="confirm-intent-email"
    >
      mock_email
    </p>
    <p
      className="css-18poma1"
    >
      PHONE
    </p>
    <p
      className="css-1n4bukm"
      data-testid="confirm-intent-phone"
    >
      mock_phone
    </p>
    <p
      className="css-18poma1"
    >
      ADDRESS
    </p>
    <p
      className="css-1n4bukm"
      data-testid="confirm-intent-address"
    >
      mock_address
    </p>
  </div>
  <section
    className="css-8atqhb"
  >
    <ul
      className="css-0"
      role="list"
    >
      <li
        className="css-tic8xi"
        role="listitem"
      >
        <img
          alt="mock_name"
          className="css-rhemcv"
        />
        <div
          className="css-1z0sfv6"
        >
          <strong
            className="css-18poma1"
          >
            mock_name
          </strong>
          <em
            className="css-183x3ib"
          />
        </div>
        <p
          className="css-15x91gx"
        >
          mock_desc
        </p>
        <strong
          className="css-nwzge5"
        >
          10.00
           $
        </strong>
      </li>
    </ul>
    <footer
      className="css-0"
    >
      <p
        className="css-ohvq3n"
      >
        TOTAL PRICE
      </p>
      <strong
        className="css-lk14dv"
      >
         $
      </strong>
    </footer>
  </section>
  <Elements
    stripe="stripe"
  >
    <div
      className="css-1mi3tt8"
    >
      <label
        className="css-8qu6k5"
        htmlFor="StripeElement"
      >
        enter your card details to finish payment
      </label>
      <div
        className="css-1dahfv9"
      >
        <CardElement
          data-testid="confirm-card-element"
          onChange={[Function]}
          options={
            Object {
              "iconStyle": "solid",
              "style": Object {
                "base": Object {
                  ":-webkit-autofill": Object {
                    "color": "#D6BCFA",
                  },
                  "::placeholder": Object {
                    "color": "#D6BCFA",
                  },
                  "color": "#FAF5FF",
                  "fontFamily": "Montserrat, Open Sans, Segoe UI, sans-serif",
                  "fontSize": "16px",
                  "fontSmoothing": "antialiased",
                  "fontWeight": 500,
                  "iconColor": "#FAF5FF",
                },
                "invalid": Object {
                  "color": "#C53030",
                  "iconColor": "#C53030",
                },
              },
            }
          }
        />
      </div>
      <p
        className="css-165naot"
      >
        Use any future date and CVC with 
        <strong>
          4242 4242 4242 4242
        </strong>
         
        number for testing
      </p>
    </div>
  </Elements>
  <div
    className="css-qv6zkf"
  >
    <button
      className="css-15ecn8p"
      data-cy="button"
      data-testid="confirm-controls-cancel"
      href="#void"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      CANCEL ORDER
    </button>
    <button
      className="css-187kagv"
      data-cy="button"
      data-testid="confirm-controls-confirm"
      disabled={true}
      href="#void"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      CONFIRM ORDER
    </button>
  </div>
</div>
`;
